### PR TITLE
Added valid allowed internal headers

### PR DESCRIPTION
**Pull request by shtripat**
Needed for replication of SSE-C objects to other sites while site replication.

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -512,6 +512,20 @@ func isAmzHeader(headerKey string) bool {
...
 }
 
+var supportedInternalHeaders = map[string]bool{
+	"x-minio-internal-server-side-encryption-sealed-key":     true,
+	"x-minio-internal-inline-data":                           true,
+	"x-minio-internal-server-side-encryption-seal-algorithm": true,
+	"x-minio-internal-server-side-encryption-iv":             true,
+	// Add more supported headers here.
+	// Must be lower case.
+}
```
Comment (harshavardhana) [inline utils.go:522]: We shouldn't mention it in this fashion, instead it must be `x-minio-replication-server-side-encryption-sealed-key` follow this for others as well. 

Convert this on the target to relevant internal values.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: utils.go (hunk 1 above)
Comment (shtripat) [reply utils.go:522]: ok, will do the changes

File: utils.go [hunk 2]
```diff
@@ -512,6 +512,20 @@ func isAmzHeader(headerKey string) bool {
 	return strings.HasPrefix(key, "x-amz-meta-") || strings.HasPrefix(key, "x-amz-grant-") || key == "x-amz-acl" || isSSEHeader(headerKey) || strings.HasPrefix(key, "x-amz-checksum-")
 }
 
+var supportedReplicationHeaders = map[string]bool{
+	"x-minio-replication-server-side-encryption-sealed-key":     true,
+	"x-minio-replication-inline-data":                           true,
```
Comment (harshavardhana) [inline utils.go:517]:
```suggestion
	"x-minio-replication-inline-data":                           true,
```

is this needed? @shtripat

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: utils.go (hunk 2 above)
Comment (shtripat) [reply utils.go:517]: may not be. will try removing and testing.

Comment (shtripat) [reply utils.go]: removed now as not needed.

File: utils.go [hunk 3]
```diff
@@ -512,6 +512,19 @@ func isAmzHeader(headerKey string) bool {
...
+var supportedReplicationHeaders = map[string]bool{
+	"x-minio-replication-server-side-encryption-sealed-key":     true,
+	"x-minio-replication-server-side-encryption-seal-algorithm": true,
+	"x-minio-replication-server-side-encryption-iv":             true,
+	// Add more supported headers here.
+	// Must be lower case.
+}
+
+// isValidReplicationHeader returns true if header is one of valid internal headers
+func isValidReplicationHeader(headerKey string) bool {
```
Comment (harshavardhana) [inline utils.go:524]:
```suggestion
func isValidReplicationEncryptionHeader(headerKey string) bool {
```

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**